datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireItemAdapterTest.kt

```diff
@@ -348,4 +354,52 @@ class QuestionnaireItemAdapterTest {
       )
       .isFalse()
   }
+
+  @Test
+  fun onCreateViewHolder_customViewType_shouldReturnCorrectCustomViewHolder() {
+    val viewFactoryMatchers = getQuestionnaireItemViewHolderFactoryMatchers()
+    val questionnaireItemAdapter = QuestionnaireItemAdapter(viewFactoryMatchers)
+    val actualQuestionnaireItemViewHolder =
+      questionnaireItemAdapter.onCreateViewHolder(
+        mock(),
+        QuestionnaireItemViewHolderType.values().size
+      )
+    assertNotNull(actualQuestionnaireItemViewHolder)
+    assertEquals(viewFactoryMatchers[0].factory.create(mock()), actualQuestionnaireItemViewHolder)
+  }
+
+  @Test(expected = QuestionnaireItemAdapter.InvalidQuestionnaireWidgetTypeException::class)
+  fun onCreateViewHolder_customViewType_shouldThrowExceptionForInvalidWidgetType() {
+    val questionnaireItemAdapter =
+      QuestionnaireItemAdapter(getQuestionnaireItemViewHolderFactoryMatchers())
+    questionnaireItemAdapter.onCreateViewHolder(
+      mock(),
+      QuestionnaireItemViewHolderType.values().size + 1
+    )
+  }
+
+  @Test
+  fun getItemViewTypeMapping_customViewType_shouldReturnCorrectIntValue() {
+    val expectedItemViewType = QuestionnaireItemViewHolderType.values().size
+    val questionnaireItemViewItem: Questionnaire.QuestionnaireItemComponent =
+      Questionnaire.QuestionnaireItemComponent()
+    questionnaireItemViewItem.type = Questionnaire.QuestionnaireItemType.DATE
+
+    val questionnaireItemAdapter =
+      QuestionnaireItemAdapter(getQuestionnaireItemViewHolderFactoryMatchers())
+    val actualItemViewType =
+      questionnaireItemAdapter.getItemViewTypeMapping(questionnaireItemViewItem)
+    assertEquals(expectedItemViewType, actualItemViewType)
+  }
+
+  private fun getQuestionnaireItemViewHolderFactoryMatchers():
+    List<QuestionnaireFragment.QuestionnaireItemViewHolderFactoryMatcher> {
+    val customQuestionnaireItemViewHolderFactory: QuestionnaireItemViewHolderFactory = mock()
+    whenever(customQuestionnaireItemViewHolderFactory.create(any())).thenReturn(mock())
```

Comment:
can you try this:

```suggestion
    val customQuestionnaireItemViewHolderFactory: QuestionnaireItemViewHolderFactory = mock().apply{
      whenever(create(any())).thenReturn(mock())
    }
```
?

and this may even allow you to inline this variable?

Reply:
done